src/presets/nodeless.ts

```diff
@@ -89,7 +89,17 @@ const nodeless: Preset & { alias: Map<string, string> } = {
     global: "unenv/runtime/node/_global",
     process: "unenv/runtime/node/process",
     Buffer: ["buffer", "Buffer"],
-    performance: "unenv/runtime/polyfill/performance",
+    performance: ["perf_hooks", "performance"],
+    Performance: ["perf_hooks", "Performance"],
+    PerformanceEntry: ["perf_hooks", "PerformanceEntry"],
+    PerformanceMark: ["perf_hooks", "PerformanceMark"],
+    PerformanceMeasure: ["perf_hooks", "PerformanceMeasure"],
+    PerformanceObserver: ["perf_hooks", "PerformanceObserver"],
+    PerformanceObserverEntryList: [
+      "perf_hooks",
+      "PerformanceObserverEntryList",
+    ],
+    PerformanceResourceTiming: ["perf_hooks", "PerformanceResourceTiming"],
```

Comment:
To be honest, I'm still not sure about this part. Initial injections in nodeless (`global`, `process`, and `Buffer`) are all Node.js specific global which were safe to inject globally.

[Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance) is a Web standard on the other hand, is not. By adding this injections we are effectively **overriding any native global implementation** in user-code.

The issue above is not easily solvable because Node.js variation of implementation needs classes with Constructor support.

I think this should be an explicit opt-in. `wrangler` preset (#262) is a good candidate because i suppose you also require explicit (Node-compat) from users to enable this and then things make sense.

Reply:
(i would suggest to revert this part and split to another PR to unblock)

Reply:
> [Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance) is a Web standard on the other hand. By adding this injections we are effectively overriding any native global implementation in user-code to unenv/node.

You make a really good point, Pooya! It's funny how we sometimes get caught up in not seeing the obvious.

I think the right thing here is to move these injects into the cloudflare preset because workerd doesn't offer Performance* APIs.

I'm not yet convinced that we need a dedicated wrangler preset. I'm hoping we really don't have to do that.

Would you be ok with me updating the PR by moving these injects into `preset/cloudflare`?

Reply:
I want to use same cloudflare preset in Nitro builds as well and users would face similar implicit opt-in to Node.js variant of `Performance*` variants if we do this for them...

Reply:
I see what you mean. The issue you are bringing up is that Node's Performance* API is not 100% compatible with Web's Performance* and if someone relies on globals we'd be making the choice for them, which might not be good. This is quite messy. How big is the difference between the two?

@jasnell as a node core contributor, have you seen this become a problem in the node community? how do people deal with these compat issues?

Reply:
- Current web impl: https://github.com/unjs/unenv/tree/main/src/runtime/web/performance
- Current node impl (extending web): https://github.com/unjs/unenv/blob/main/src/runtime/node/perf_hooks/internal/performance.ts

Difference is:
- In node impl, we always implement class and won't fall back to `globalThis.*` natives if available 
- Node.js classes have additional props

The first is solvable but I'm more worried about the design decisions we hare making for unenv. For me unless explicitly we are adding "node.js" compatibility, we should strictly stick with Web standards spec. Only for wrangler usage that uses unenv as Node.js compatibility layer it clearly makes sense but for generic polyfill, it breaks the concept.

I am thinking to expose new API from unenv that we can have flags like `nodeCompat` to make difference behavior.


---

In the meantime, let's move these to cloudflare as you suggested this PR seems unnecessarily being blocked.

Reply:
we do have `performance.now()` and `performance.timeOrigin` in workers. The impl in Node.js definitely has a bunch of Node.js specific stuff. It hasn't been too much of a problem in the ecosystem but it definitely warrants caution.